scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,278 @@
+  width: "100%",
+  height: "100%",
+  maxHeight: "inherit",
+};
+
+const imageContentStyle: React.CSSProperties = {
+  width: "100%",
+  maxHeight: "inherit",
+  overflowY: "auto",
+  display: "grid",
+  gridRowGap: "4px",
+};
+
+const imageStyle: React.CSSProperties = {
+  width: "100%",
+  height: "auto",
+};
+
+const imageWrapperStyle: React.CSSProperties = {
+  cursor: "pointer",
+  maxHeight: "100%",
+};
+
+const selectedCardStyle: React.CSSProperties = {
+  display: "flex",
+  flexDirection: "column",
+  height: "100%",
+  overflow: "hidden",
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "4px",
+};
+
+const cardStyle: React.CSSProperties = {
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "2px",
+};
+
+let isMounted: boolean = false;
+
+const ImageTaggingComponent = (props: IProps) => {
+  const { superdesk, data, style } = props;
+  const { httpRequestJsonLocal } = superdesk;
+
+  const [showImages] = React.useState<boolean>(true);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [fetchError, setFetchError] = React.useState<boolean>(false);
+  const [selectedImage, setSelectedImage] = React.useState<IImage | null>(null);
+  const [images, setImages] = React.useState<Array<IImage>>([]);
+
+  const fetchImages = async (item: Array<ITag>, signal: AbortSignal) => {
+    setIsLoading(true);
+    httpRequestJsonLocal({
+      abortSignal: signal,
+      method: "POST",
+      path: "/ai_image_suggestions/",
+      payload: {
+        service: "imatrics",
+        item,
+      },
+    })
+      .then((res: any) => {
+        if (isMounted) {
+          setFetchError(false);
+          try {
+            setSelectedImage(res.result[0]);
+            setImages(res.result ?? []);
+          } catch {
+            setSelectedImage(null);
+            setImages([]);
+          }
+        }
+      })
+      .catch(() => {
+        setFetchError(true);
+      })
+      .finally(() => isMounted && setIsLoading(false));
+  };
+
+  const formatTags = (concepts: any) => {
+    let res: Array<ITag> = [];
+    for (const key in concepts) {
+      if (key === "subject") {
+        concepts[key].forEach((concept: any) => {
+          res.push({
+            title: concept.name,
+            type: "category",
+            pubStatus: true,
+            weight: 1,
+          });
+        });
+      } else {
+        concepts[key].forEach((concept: any) => {
+          res.push({
+            title: concept.name,
+            type: key,
+            pubStatus: true,
+            weight: 1,
+          });
+        });
+      }
+    }
+    return res;
+  };
+
+  const handleClickImage = (image: IImage) => {
+    setSelectedImage(image);
+  };
+
+  const handleDragStart = (
+    event: React.DragEvent<HTMLDivElement>,
+    image: any,
+  ) => {
+    try {
+      event.dataTransfer.setData(
+        "application/superdesk.item.picture",
+        JSON.stringify({
```

Comment:
Specify the type of an object as `IArticle` before passing to `JSON.stringify`. This allows for TypeScript compiler to validate the properties.

```
const item: IArticle = {...};

event.dataTransfer.setData(
        "application/superdesk.item.picture",
        JSON.stringify(item),
)
```

Reply:
This is solved. It was hard to guess the payload, so I reused the article and changed what seemed relevant fields. Thanks for the feedback!